18_migrate_ci_to_gh_actions.md

```diff
@@ -0,0 +1,212 @@
+- Once the above is complete: Stop using Travis on Keylime repos completely.
+
+### Non-Goals
+
+<!--
+What is out of scope for this enhancement?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+Keylime's current Travis config is located, as all Travis configs are, in
+`~/.travis.yml`. All fundamental functionality of this config should be
+replicated in an equivalent Actions workflow, which would be housed under, for
+example, `~/.github/workflows/test.yml`.
+
+The current config makes use of several containers. It will be worth
+investigating whether these containers can be run on Github's standard runners,
+and how we might have to change the containers to fully migrate (if at all).
+
+### User Stories (optional)
+
+<!--
+Detail the things that people will be able to do if this enhancement is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+#### Story 2
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+enhancement ecosystem.
+
+How will security be reviewed and by whom?
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy.  Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+-->
+
+### Upgrade / Downgrade Strategy
+
+Github Actions configs can be updated by simply changing their workflow file in
+a pull request.
+
+Changes to existing workflows via PR will be reflected in the workflow that
+runs on the PR that makes the changes. This is particularly handy for code
+tests.
+
+## Drawbacks
+
+<!--
+Why should this enhancement _not_ be implemented?
+-->
+
+## Alternatives
+
```

Comment:
You mentioned above some alternatives to Travis but didn't list them. Not too important, but maybe a short list here would work.